server/src/test/java/org/elasticsearch/index/query/SpanTermQueryBuilderTests.java

```diff
@@ -80,16 +80,20 @@ protected void doAssertLuceneQuery(SpanTermQueryBuilder queryBuilder, Query quer
      * @return an array of random {@link SpanTermQueryBuilder} with same field name
      */
     public SpanTermQueryBuilder[] createSpanTermQueryBuilders(int amount) {
+        return createSpanTermQueryBuilders(amount, true, true);
+    }
+
+    public SpanTermQueryBuilder[] createSpanTermQueryBuilders(int amount, boolean supportsBoost, boolean supportsQueryName) {
```

Comment:
I have noticed that you always call `createSpanTermQueryBuilders(_, false, true)` with `supportsBoost=false` and `supportQueryName=true`.  I don't think we need these parameters at all.  Looks like this method is only called from complex span queries, so you can just assume that we don't set boosting here, and just remove  the lines where the boost is set:
```
     spanTermQuery.boost(2.0f / randomIntBetween(1, 20));
```

Reply:
The `span_first` query supports boost and we use the boost and query name [here](https://github.com/elastic/elasticsearch/pull/34112/files/0ac11ecd4fda4feab10d13f0b1595e35d8f79df0#diff-fd26d71bc739f4672fae13a9d0a32a56R83).

Query name is always `true`, so we can remove this parameter, here is a commit to do so f2861cdae54126cac275d3eaf86b72acd47cd376.

Reply:
@cbismuth  can you please explain why `span_first` has a special treatment?  Shouldn't we disallow for its single inner span query to have `boost` as well?

Reply:
Totally. It has been removed in c96aaa9aa5c25f0473f7e702a799034557ba06d1.